### PR TITLE
Added changes to support Azure functions

### PR DESCRIPTION
**Pull request by maliksahil**
Changes are in AzureFunctionsAuthenticationHttpContextExtension.cs.

### CODE REVIEW

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        /// </summary>
+        /// <param name="httpContext">The current HTTP Context, such as req.HttpContext.</param>
+        /// <returns>A task indicating success or failure. In case of failure <see cref="Microsoft.Identity.Web.UnauthorizedObjectResult"/>.</returns>
+        public static async Task<(bool, IActionResult?)> AuthenticateAzureFunctionAsync(
+            this HttpContext httpContext)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException("Parameter httpContext cannot be null");
+            }
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:28]: nit: `throw new ArgumentNullException(nameof(httpContext));`

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+            AuthenticateResult? result =
+                await httpContext.AuthenticateAsync("Bearer").ConfigureAwait(false);
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:31]: >"Bearer" [](start = 52, length = 8)

we have a constant value-> `Constants.Bearer`

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            if (result.Succeeded)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:32]: add a null check here?

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs (hunk 1 above)
Comment (maliksahil) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:28]: Addressed.

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs (hunk 2 above)
Comment (maliksahil) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs]: Addressed

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs (hunk 3 above)
Comment (maliksahil) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs]: Addressed

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Extensions for <see cref="AzureFunctionsAuthenticationHttpContextExtension"/>.
+    /// </summary>
+    public static class AzureFunctionsAuthenticationHttpContextExtension
+    {
+        /// <summary>
+        /// Enables Bearer authentication for an API for use in Azure Functions.
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:18]: What about?
Enables an Azure Function to act as/expose a protected web API, enabling bearer token authentication? Calling this method from your Azure function validates the token and exposes the identity of the user or app on behalf of which your function is called, in the `HttpContext.User` member, where your function can exploit it.

**Review by jmprieur — COMMENTED** [review]
Thanks @maliksahil 
LGTM
- I suggested a change in the XML comment for the method.
- @jennyf19, I think we should also add the test app?

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs (hunk 4 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:18]: @jmprieur what do you mean by add the test app?

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs (hunk 4 above)
Comment (maliksahil) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:18]: Hmm PR is approved, should I still go ahead and edit this description? New PR or same?

Comment (jennyf19) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs:18]: @maliksahil go ahead and edit. as it's a fork, @jmprieur  or i will have to merge it. thanks.

Comment (maliksahil) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs]: Done!! :-)

Comment (jmprieur) [reply src/Microsoft.Identity.Web/AzureFunctionsAuthenticationHttpContextExtension.cs]: @jennyf19, I meant the sample using the new API: https://github.com/maliksahil/ms-identity-azurefunctions-microsoft-identity-web

**Pull request merged**